enterprise_access/apps/bffs/response_builder.py

```diff
@@ -84,18 +84,22 @@ def serialize(self):
         if not hasattr(self, 'serializer_class') or self.serializer_class is None:
             raise NotImplementedError("Subclasses must define a serializer_class.")
 
+        serializer = self.serializer_class(data=self.response_data)
+
         try:
-            serializer = self.serializer_class(data=self.response_data)
             serializer.is_valid(raise_exception=True)
-            return serializer.data, self.status_code
         except Exception as exc:  # pylint: disable=broad-except
             logger.exception('Could not serialize the response data.')
             self.context.add_warning(
                 user_message='An error occurred while processing the response data.',
                 developer_message=f'Could not serialize the response data. Error: {exc}',
             )
-            self.add_errors_warnings_to_response()
-            return self.serializer_class(self.response_data).data, self.status_code
+
+        serialized_data = serializer.data
+        serialized_data['errors'] = self.context.errors
+        serialized_data['warnings'] = self.context.warnings
+        serialized_data['enterprise_features'] = getattr(self.context, 'enterprise_features', {})
+        return serialized_data, self.status_code
```

Comment:
ok, this is the added bug fix. previously, if errors occurred outside of serialization (like in the handler), but the serialization still ended up being valid, we'd lose track of the errors/warnings.

Reply:
It feels weird that `add_errors_warnings_to_response()` exists but we're still duplicating that logic here?  Is it possible to just call that function before serialization?